pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -623,8 +691,132 @@ func (r *ReconcileSyncSetInstance) applySyncSetPatches(ssi *hivev1.SyncSetInstan
+		for _, rss := range ssi.Status.SecretReferences {
+			if rss.Name == secretReferenceSyncStatus.Name &&
+				rss.Namespace == secretReferenceSyncStatus.Namespace &&
+				rss.APIVersion == secretReferenceSyncStatus.APIVersion &&
+				rss.Kind == secretReferenceSyncStatus.Kind {
+				secretReferenceSyncConditions = rss.Conditions
+				found = true
+				if rss.Hash != secretReferenceSyncStatus.Hash {
+					ssiLog.Debugf("Secret %s/%s (%s) has changed, will re-apply", secretReference.Target.Namespace, secretReference.Target.Name, secretReference.Target.Kind)
+					different = true
+					break
+				}
+
+				// re-apply if failure occurred
+				if failureCondition := controllerutils.FindSyncCondition(rss.Conditions, hivev1.ApplyFailureSyncCondition); failureCondition != nil {
+					if failureCondition.Status == corev1.ConditionTrue {
+						ssiLog.Debugf("Secret %s/%s (%s) failed last time, will re-apply", secretReference.Target.Namespace, secretReference.Target.Name, secretReference.Target.Kind)
+						shouldReApply = true
+						break
+					}
+				}
+
+				// re-apply if two hours have passed since LastProbeTime
+				if applySuccessCondition := controllerutils.FindSyncCondition(rss.Conditions, hivev1.ApplySuccessSyncCondition); applySuccessCondition != nil {
+					since := time.Since(applySuccessCondition.LastProbeTime.Time)
+					if since > reapplyInterval {
+						ssiLog.Debugf("It has been %v since secret %s/%s (%s) was last applied, will re-apply", since, secretReference.Target.Namespace, secretReference.Target.Name, secretReference.Target.Kind)
+						shouldReApply = true
+					}
+				}
+				break
+			}
+		}
+
+		if !found || different || shouldReApply {
+			if err := apis.AddToScheme(scheme.Scheme); err != nil {
+				return err
+			}
+			accessor, err := meta.Accessor(secret)
+			if err != nil {
+				ssiLog.WithError(err).Errorf("cannot create accessor for secret")
+				return err
+			}
+			accessor.SetName(secretReference.Target.Name)
+			accessor.SetNamespace(secretReference.Target.Namespace)
+
+			ssiLog.Debugf("applying secret: %s/%s (%s)", secret.Namespace, secret.Name, secret.Kind)
+			var result hiveresource.ApplyResult
+			result, applyErr = h.ApplyRuntimeObject(secret, scheme.Scheme)
+			secretReferenceSyncStatus.Conditions = r.setApplySyncConditions(secretReferenceSyncConditions, applyErr)
+			if applyErr != nil {
+				ssiLog.WithError(applyErr).Errorf("error applying secret %s/%s (%s)", secret.Namespace, secret.Name, secret.Kind)
+			} else {
+				ssiLog.Debugf("resource %s/%s (%s): %s", secret.Namespace, secret.Name, secret.Kind, result)
+			}
+		} else {
+			ssiLog.Debugf("resource %s/%s (%s) has not changed, will not apply", secret.Namespace, secret.Name, secret.Kind)
+			secretReferenceSyncStatus.Conditions = secretReferenceSyncConditions
+		}
+
+		syncStatusList = append(syncStatusList, secretReferenceSyncStatus)
+
+		// If an error applying occurred, stop processing right here
+		if applyErr != nil {
+			break
```

Comment:
Why would we stop processing the rest of the secrets if there was an error with one of them? We should be aggregating errors together instead of bailing out at the first error.

Reply:
Asked about this in standup today. Consensus was that we were failing at first error by design to mimic other config management systems and that if users would like resources to be created separately then those resources should be in separate syncsets.